frontend/src/app/components/crm/leads/leads.component.ts

```diff
@@ -90,42 +90,32 @@ export class LeadsComponent implements AfterViewInit {
        * @param filter The value that the user searches transformedFilter.
        */
       this.dataSource.filterPredicate = (data, filter: string)  => {
-        const accumulator = (currentTerm, key) => {
-          //Using the recusrsive method here becuase using an if/else condition will increase space complexity n times.
-          return this.nestedPropertyFilterCheck(currentTerm, data, key);
+        let cleanString = (str: string): string => str.trim().toLowerCase();
+        let hasFilter = (data: object | number | string | boolean, filter: string): boolean => {
+          // traverse through JSON's tree like structure
+          if(typeof data === 'object') {
+            let keys = Object.keys(data);
+            for(let key in keys) {
```

Comment:
can key be "const" instead of "let"? ditto elsewhere when it's not being changed.